tests/mocha/json_test.js

```diff
@@ -0,0 +1,343 @@
+      var VALUE1 = 'VALUE1';
+      Blockly.defineBlocksWithJsonArray([{
+        "type": BLOCK_TYPE,
+        "message0": "%1",
+        "args0": [
+          {
+            "type": "field_dropdown",
+            "name": FIELD_NAME,
+            "options": [
+              [LABEL0, VALUE0],
+              [LABEL1, VALUE1]
+            ]
+          }
+        ]
+      }]);
+
+      var block = new Blockly.Block(this.workspace_, BLOCK_TYPE);
+      this.blocks_.push(block);
+      assertEquals(1, block.inputList.length);
+      assertEquals(1, block.inputList[0].fieldRow.length);
+      var dropdown = block.inputList[0].fieldRow[0];
+      assertEquals(dropdown, block.getField(FIELD_NAME));
+      assertEquals(Blockly.FieldDropdown, dropdown.constructor);
+      assertEquals(VALUE0, dropdown.getValue());
+
+      var options = dropdown.getOptions();
+      assertEquals(LABEL0, options[0][0]);
+      assertEquals(VALUE0, options[0][1]);
+      assertEquals(LABEL1, options[1][0]);
+      assertEquals(VALUE1, options[1][1]);
+    });
+
+
+    test('Dropdown with images', function() {
+      var BLOCK_TYPE = 'test_json_dropdown';
+      this.blockTypes_.push(BLOCK_TYPE);
+      var FIELD_NAME = 'FIELD_NAME';
+      var IMAGE1_ALT_TEXT = 'Localized message.';
+      Blockly.Msg['ALT_TEXT'] = IMAGE1_ALT_TEXT;
+      this.messages_.push('ALT_TEXT');
+      var IMAGE0 = {
+        'width': 12,
+        'height': 34,
+        'src': 'http://image0.src',
+        'alt': 'IMAGE0 alt text'
+      };
+      var VALUE0 = 'VALUE0';
+      var IMAGE1 = {
+        'width': 56,
+        'height': 78,
+        'src': 'http://image1.src',
+        'alt': '%{BKY_ALT_TEXT}'
+      };
+      var VALUE1 = 'VALUE1';
+      var IMAGE2 = {
+        'width': 90,
+        'height': 123,
+        'src': 'http://image2.src'
+      };
+      var VALUE2 = 'VALUE2';
+
+      Blockly.defineBlocksWithJsonArray([{
+        "type": BLOCK_TYPE,
+        "message0": "%1",
+        "args0": [
+          {
+            "type": "field_dropdown",
+            "name": FIELD_NAME,
+            "options": [
+              [IMAGE0, VALUE0],
+              [IMAGE1, VALUE1],
+              [IMAGE2, VALUE2]
+            ]
+          }
+        ]
+      }]);
+
+      var block = new Blockly.Block(this.workspace_, BLOCK_TYPE);
+      this.blocks_.push(block);
+      assertEquals(1, block.inputList.length);
+      assertEquals(1, block.inputList[0].fieldRow.length);
+      var dropdown = block.inputList[0].fieldRow[0];
+      assertEquals(dropdown, block.getField(FIELD_NAME));
+      assertEquals(Blockly.FieldDropdown, dropdown.constructor);
+      assertEquals(VALUE0, dropdown.getValue());
+
+      var options = dropdown.getOptions();
+      var image0 = options[0][0];
+      assertEquals(IMAGE0.width, image0.width);
+      assertEquals(IMAGE0.height, image0.height);
+      assertEquals(IMAGE0.src, image0.src);
+      assertEquals(IMAGE0.alt, image0.alt);
+      assertEquals(VALUE0, options[0][1]);
```

Comment:
Could this be pulled out into a helper assert functino?

Reply:
Done.